src/object/cli_obj.c

```diff
@@ -568,9 +568,10 @@ obj_ec_leader_select(struct dc_object *obj, int grp_idx, bool cond_modify, uint3
 	 * If no parity node is available, then handle related task that has conditional
 	 * modification via distributed tranasaction.
 	 */
-	if (cond_modify)
+	if (cond_modify) {
+		D_RWLOCK_UNLOCK(&obj->cob_lock);
 		return -DER_NEED_TX;
-
+	}
```

Comment:
Do we want to use `D_GOTO(unlock, -DER_NEED_TX)` to be consistent in this function?

Reply:
Thank you! Just updated as you suggested.